reactor-core/src/main/java/reactor/core/scheduler/ElasticScheduler.java

```diff
@@ -300,6 +308,13 @@ public void run() {
 				cached.dispose();
 			}
 		}
+
+		@Override
+		public void dispose() {
```

Comment:
perhaps we should do it in `SchedulerTask`? `directSchedule` returns it, and we can handle the disposed state there

/cc @simonbasle

Reply:
I'll try. I put it here because `SchedulerTask` seemed to have a bit more complex state.

Reply:
`SchedulerTask` is also used in the other schedulers that do not manage executor pool. I don't see much benefit adding a reference/handling of `Disposable` in `SchedulerTask` just for the `ElasticScheduler`. Maybe we can copy `SchedulerTask` into `ElasticScheduler` and specialize for it.

Reply:
`Disposables.composite` allocates an object and it is a part of the hot path, I thought about avoiding it